### PR TITLE
Change camel version to 3.8.0

### PR DESCRIPTION
**Pull request by zimmerml**
* Change camel to version 3.8.0
* Adapt dependencies
* Remove legacy code

closes #141, closes #142, closes #143, closes #144, closes #145, closes #146

### CODE REVIEW

File: org.opentosca.bus/org.opentosca.bus.application.api.jsonhttp/src/main/java/org/opentosca/bus/application/api/jsonhttp/servicehandler/ApplicationBusServiceHandler.java [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
-import org.springframework.stereotype.Service;
-
-/**
- * Service Handler of the Application Bus-JSON/HTTP-API.<br>
- * <br>
- * <p>
- * Here the implementation of the IApplicationBusService is binded or unbinded. During the binding the routing endpoint
- * of the Application Bus is handed over.
- *
- * @author Michael Zimmermann - zimmerml@studi.informatik.uni-stuttgart.de
```
Comment (nyuuyn) [inline org.opentosca.bus/org.opentosca.bus.application.api.jsonhttp/src/main/java/org/opentosca/bus/application/api/jsonhttp/servicehandler/ApplicationBusServiceHandler.java:16]: Does removing this not kill the App Bus ?

File: org.opentosca.bus/org.opentosca.bus.application.api.soaphttp/src/main/java/org/opentosca/bus/application/api/soaphttp/route/Route.java [hunk 2]
```diff
@@ -41,26 +37,19 @@ public void configure() throws Exception {
...
 
         final Processor requestProcessor = new RequestProcessor();
         final Processor responseProcessor = new ResponseProcessor();
 
-        from(SOAP_ENDPOINT).unmarshal(jaxb).process(requestProcessor).choice().when(APP_BUS_ENDPOINT_EXISTS)
-            //FIXME this recipientList should be replaced with a directly injected service reference
-            .recipientList(APP_BUS_ENDPOINT).to("direct:handleResponse").endChoice().otherwise()
-            .to("direct:handleException");
+        from(SOAP_ENDPOINT).unmarshal(jaxb).process(requestProcessor)
+                           .to("direct-vm:org.opentosca.bus.application.service").to("direct:handleResponse");
```
Comment (nyuuyn) [inline org.opentosca.bus/org.opentosca.bus.application.api.soaphttp/src/main/java/org/opentosca/bus/application/api/soaphttp/route/Route.java:48]: the recipient list is gone? wasn't there a problem in the past, which was elegantly solved with the list ?

File: org.opentosca.bus/org.opentosca.bus.management.api.resthttp/pom.xml [hunk 3]
```diff
@@ -1,45 +1,47 @@
...
+		</dependency>
+		<dependency>
+			<groupId>org.apache.camel</groupId>
+			<artifactId>camel-core</artifactId>
+			<version>${camel.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>org.restlet.jee</groupId>
+			<artifactId>org.restlet</artifactId>
+			<version>2.3.6</version>
```
Comment (nyuuyn) [inline org.opentosca.bus/org.opentosca.bus.management.api.resthttp/pom.xml:29]: maybe pull it into a maven propertiy

File: org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.soaphttp/src/main/java/org/opentosca/bus/management/invocation/plugin/soaphttp/processor/CallbackProcessor.java [hunk 4]
```diff
@@ -45,11 +46,16 @@
...
+
+        final String message = exchange.getIn().getBody(String.class);
+        LOG.debug("Received message as callback: {}", message);
+
         final Set<String> messageIDs = ManagementBusInvocationPluginSoapHttp.getMessageIDs();
         LOG.debug("Stored messageIDs: {}", messageIDs.toString());
 
         // copy SOAP headers in camel exchange header
-        @SuppressWarnings("unchecked") final List<SoapHeader> soapHeaders = (List<SoapHeader>) exchange.getIn().getHeader(Header.HEADER_LIST);
+        @SuppressWarnings("unchecked")
```
Comment (nyuuyn) [inline org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.soaphttp/src/main/java/org/opentosca/bus/management/invocation/plugin/soaphttp/processor/CallbackProcessor.java:57]: i never understood why we do this

**Review by nyuuyn — COMMENTED** [review]
lgtm

File: org.opentosca.bus/org.opentosca.bus.application.api.jsonhttp/src/main/java/org/opentosca/bus/application/api/jsonhttp/servicehandler/ApplicationBusServiceHandler.java (hunk 1 above)
Comment (zimmerml) [reply org.opentosca.bus/org.opentosca.bus.application.api.jsonhttp/src/main/java/org/opentosca/bus/application/api/jsonhttp/servicehandler/ApplicationBusServiceHandler.java:16]: Legacy code from previous refactorings. It wasn't used anymore.

File: org.opentosca.bus/org.opentosca.bus.application.api.soaphttp/src/main/java/org/opentosca/bus/application/api/soaphttp/route/Route.java (hunk 2 above)
Comment (zimmerml) [reply org.opentosca.bus/org.opentosca.bus.application.api.soaphttp/src/main/java/org/opentosca/bus/application/api/soaphttp/route/Route.java:48]: It should be working like this now.

**Review by wederbn — APPROVED** [review]
LGTM

**Review by nyuuyn — APPROVED** [review]
lgtm

**Pull request merged**